Index: src/main/java/org/generation/italy/esempiCorso/inheritance/examples/interfaces/HumanBeing.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/org/generation/italy/esempiCorso/inheritance/examples/interfaces/HumanBeing.java b/src/main/java/org/generation/italy/esempiCorso/inheritance/examples/interfaces/HumanBeing.java
new file mode 100644
--- /dev/null	(date 1728907264418)
+++ b/src/main/java/org/generation/italy/esempiCorso/inheritance/examples/interfaces/HumanBeing.java	(date 1728907264418)
@@ -0,0 +1,8 @@
+package org.generation.italy.esempiCorso.inheritance.examples.interfaces;
+
+public interface HumanBeing {
+    int maxAge = 120; //ciò che capisce il compilatore è " public static final int max age", ovvero questo valore sarà static e una costante;
+    void speak(); //tutti i metodi di un interfaccia sono public e devono essere abstract; quindi non facciamo "public abstract void speak" perchè sarebbe ridondante, è cosi e basta.
+    void eat();
+    void work(int hours); //possiamo metterci tutti i metodi astratti che vogliamo;
+}
