Index: src/main/java/org/generation/italy/esempiCorso/inheritance/Start.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package org.generation.italy.esempiCorso.inheritance;\r\n\r\nimport org.generation.italy.esempiCorso.inheritance.school.Teacher;\r\n\r\npublic class Start {\r\n    public static void main(String[] args){\r\n       Teacher teacher = new Teacher(\"Riccardo\",\"Audano\",18, \"programmazione\", 18);\r\n       System.out.println(teacher.toString()); //tutti gli oggetti creati derivano dalla super classe object\r\n\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/org/generation/italy/esempiCorso/inheritance/Start.java b/src/main/java/org/generation/italy/esempiCorso/inheritance/Start.java
--- a/src/main/java/org/generation/italy/esempiCorso/inheritance/Start.java	(revision 344f9bb7d959550b804044b3853f4f9337584afb)
+++ b/src/main/java/org/generation/italy/esempiCorso/inheritance/Start.java	(date 1728552921343)
@@ -1,11 +1,8 @@
 package org.generation.italy.esempiCorso.inheritance;
 
-import org.generation.italy.esempiCorso.inheritance.school.Teacher;
-
 public class Start {
     public static void main(String[] args){
-       Teacher teacher = new Teacher("Riccardo","Audano",18, "programmazione", 18);
-       System.out.println(teacher.toString()); //tutti gli oggetti creati derivano dalla super classe object
-
+        System.out.println("testing github");
+        System.out.println("Aggiunto da Alessia");
     }
 }
